### PR TITLE
LWC bug fixes

### PR DESCRIPTION
**Pull request by lcampos**
### What does this PR do?
This PR fixes the following issues:
- Receiving events without data to process
- Persisting custom component information when the file path does not exist
- Persisting custom component information after indexing a project
- Not showing correct custom component information when triggering onCompletion for javascript lwc files

### What issues does this PR fix or reference?
@W-7971496@

### CODE REVIEW

File: packages/lwc-language-server/src/component-indexer.ts [hunk 1]
```diff
@@ -103,7 +112,8 @@ export default class ComponentIndexer extends BaseIndexer {
     }
 
     persistCustomComponents() {
-        const indexPath = join(this.workspaceRoot, CUSTOM_COMPONENT_INDEX_FILE);
+        const indexPath = path.join(this.workspaceRoot, CUSTOM_COMPONENT_INDEX_FILE);
```
Comment (lcampos) [inline packages/lwc-language-server/src/component-indexer.ts:115]: This was failing on projects that didn't previously had `.sfdx/indexes/lwc/` created.

File: packages/lwc-language-server/src/lwc-server.ts [hunk 2]
```diff
@@ -118,6 +127,18 @@ export default class Server {
         if (await this.context.isLWCTemplate(doc)) {
             this.auraDataProvider.activated = false; // provide completions for lwc components in an Aura template
             this.lwcDataProvider.activated = true;
+        } else if (await this.context.isLWCJavascript(doc)) {
```
Comment (lcampos) [inline packages/lwc-language-server/src/lwc-server.ts:130]: If the completion event is triggered from a lwc js file we get the already cached tag info and add their `c/<component name>` info to the rest of the suggestions provided by the html lang server.

File: packages/lwc-language-server/src/tag.ts [hunk 3]
```diff
@@ -32,7 +32,7 @@ export default class Tag implements ITagData {
         this.updatedAt = attributes.updatedAt ? new Date(attributes.updatedAt) : null;
```
Comment (rcoringrato-sfdc) [inline packages/lwc-language-server/src/tag.ts:32]: This assignment seems redundant.

Comment (lcampos) [reply packages/lwc-language-server/src/tag.ts:32]: Yeah, let me remove that line

**Pull request merged**